great_expectations/dataset/sqlalchemy_dataset.py

```diff
@@ -2093,14 +2098,15 @@ def _get_dialect_like_pattern_expression(self, column, like_pattern, positive=Tr
 
         try:
             # Bigquery
-            if isinstance(
-                self.sql_engine_dialect, pybigquery.sqlalchemy_bigquery.BigQueryDialect
-            ):
+            if hasattr(self.sql_engine_dialect, "BigQueryDialect"):
                 dialect_supported = True
         except (
             AttributeError,
             TypeError,
         ):  # TypeError can occur if the driver was not installed and so is None
+            logger.debug(
+                "Unable to load BigQueryDialect dialect while running expect_column_values_to_not_match_regex_list"
+            )
```

Comment:
Is this log description accurate? I see it referencing `expect_column_values_to_not_match_regex_list` but I imagine that other expectations could be calling this method `_get_dialect_like_pattern_expression` as well. Might also help to include the exception in the log. I suggest the following:

```suggestion
            logger.debug(
                "Unable to load BigQueryDialect dialect while running _get_dialect_like_pattern_expression",
                exc_info=True,
            )
```